packages/create-llama/templates/types/streaming/nextjs/app/api/chat/route.o11y.ts

```diff
@@ -0,0 +1,94 @@
+import * as traceloop from "@traceloop/node-server-sdk";
+import { StreamingTextResponse } from "ai";
+import * as LlamaIndex from "llamaindex";
+import { ChatMessage, MessageContent, OpenAI } from "llamaindex";
+import { NextRequest, NextResponse } from "next/server";
+import { createChatEngine } from "./engine";
+import { LlamaIndexStream } from "./llamaindex-stream";
+
+traceloop.initialize({
+  appName: "llama-app",
+  disableBatch: true,
+  instrumentModules: {
+    llamaIndex: LlamaIndex,
+  },
+});
```

Comment:
This snippet is the only difference to `route.ts`.

How about the following:

1. Adding the snippet to a new file:
`templates/components/observability/typescript/traceloop/index.ts` and wrap it in a function `initObservability`
2. Add a non-op `initObservability` function to the main Nextjs template
3. If the user selects traceloop observability, copy the `templates/components/observability/typescript/traceloop` folder

This is the same concept that is used with the different chat engines - you can have a look at the code there for reference

Reply:
Another advantage is that you can also copy this component to the two express templates

Reply:
Done!